aws/resource_aws_pinpoint_app_test.go

```diff
@@ -0,0 +1,249 @@
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).pinpointconn
+
+		// Check if the app exists
+		params := &pinpoint.GetAppInput{
+			ApplicationId: aws.String(rs.Primary.ID),
+		}
+		_, err := conn.GetApp(params)
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+const testAccAWSPinpointAppConfig_withGeneratedName = `
+provider "aws" {
+	region = "us-east-1"
+}
+
+resource "aws_pinpoint_app" "test_app" {}
+`
+
+func testAccAWSPinpointAppConfig_CampaignHookLambda(appName, funcName string) string {
+	return fmt.Sprintf(`
+provider "aws" {
+	region = "us-east-1"
+}
+
+
+resource "aws_pinpoint_app" "test_app" {
+    name = "%s"
+
+    campaign_hook {
+        lambda_function_name = "${aws_lambda_function.test.arn}"
+        mode                 = "DELIVERY"
+    }
+}
+
+resource "aws_lambda_function" "test" {
+    filename      = "test-fixtures/lambdapinpoint.zip"
+    function_name = "%s"
+    role          = "${aws_iam_role.iam_for_lambda.arn}"
+    handler       = "lambdapinpoint.handler"
+    runtime       = "nodejs6.10"
+    publish       = true
+}
+
+resource "aws_iam_role" "iam_for_lambda" {
+    name = "test-role"
+    assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+data "aws_caller_identity" "aws" {}
+
+resource "aws_lambda_permission" "permission" {
+  statement_id = "AllowExecutionFromPinpoint"
+  action = "lambda:InvokeFunction"
+  function_name = "${aws_lambda_function.test.function_name}"
+  principal = "pinpoint.us-east-1.amazonaws.com"
+  source_arn = "arn:aws:mobiletargeting:us-east-1:${data.aws_caller_identity.aws.account_id}:/apps/*"
+}
+`, appName, funcName)
+}
+
+const testAccAWSPinpointAppConfig_Limits = `
+provider "aws" {
+	region = "us-east-1"
+}
+
+resource "aws_pinpoint_app" "test_app" {
+    name = "terraform-test-pinpointapp-limits"
+
+    limits {
+        daily               = 3
+        maximum_duration    = 600
+        messages_per_second = 50
+        total               = 500
+    }
+}
+`
+
+const testAccAWSPinpointAppConfig_QuietTime = `
+provider "aws" {
+	region = "us-east-1"
+}
+
+resource "aws_pinpoint_app" "test_app" {
+    name = "terraform-test-pinpointapp-quiet"
+
+    quiet_time {
+        start = "00:00"
+        end   = "03:00"
+    }
+}
+`
+
+func testAccCheckAWSPinpointAppDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).pinpointconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_pinpoint_app" {
+			continue
+		}
+
+		// Check if the topic exists by fetching its attributes
+		params := &pinpoint.GetAppInput{
+			ApplicationId: aws.String(rs.Primary.ID),
+		}
+		_, err := conn.GetApp(params)
+		if err != nil {
+			if isAWSErr(err, pinpoint.ErrCodeNotFoundException, "") {
+				return nil
```

Comment:
Nitpick: While generally the testing configuration will only include one `aws_pinpoint_app`, we should use `continue` here instead of `return nil` in case it includes multiple applications. 👍

Reply:
Fixed in https://github.com/terraform-providers/terraform-provider-aws/pull/5956/commits/be7dc0047e5e3ec5d966f2a51c702546a27cfb95